src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -665,6 +667,8 @@ export class TerminalInstance implements ITerminalInstance {
 		const env = shell.env ? shell.env : TerminalInstance._cloneEnv(parentEnv);
 		env['PTYPID'] = process.pid.toString();
 		env['PTYSHELL'] = shell.executable;
+		env['TERM_PROGRAM'] = product.nameShort;
```

Comment:
I think just a consistent 'vscode' across all would be best as this is meant for scripts to be able to see what the terminal is (maybe certain ones have more capabilities). `TERM_PROGRAM` is good for differentiating versions if needed.